src/main/treeDataProviders/issuesTree/issuesTreeDataProvider.ts

```diff
@@ -336,38 +359,69 @@ export class IssuesTreeDataProvider implements vscode.TreeDataProvider<IssuesRoo
-                                workspaceData.descriptorsIssuesData.push(descriptorData);
-                                root.addChildAndApply(descriptorWithIssues);
-                                progressManager.onProgress();
-                            }
-                        })
-                        .catch(error => this.onFileScanError(workspaceData, root, error, descriptorData))
-                        .finally(() => progressManager.onProgress())
+                    this.createDescriptorTask(workspaceData, root, descriptorData, descriptorNode, descriptorGraph, progressManager, checkCanceled)
                 );
             }
         }
         await Promise.all(scansPromises);
     }
 
+    /**
+     * Runs the descriptor scans async
+     * 1. dependency graph scan
+     * 2. applicable cve scan
+     * @param workspaceData - the issues data for the workspace
+     * @param root - the root node of the workspace
+     * @param descriptorData - the descriptor issues data
+     * @param descriptorNode - the descriptor node
+     * @param descriptorGraph - the descriptor dependencies graph
+     * @param progressManager - the progress manager for the process
+     * @param checkCanceled - the method to check if cancle was requested
+     */
+    private async createDescriptorTask(
+        workspaceData: WorkspaceIssuesData,
+        root: IssuesRootTreeNode,
+        descriptorData: DescriptorIssuesData,
+        descriptorNode: DescriptorTreeNode,
+        descriptorGraph: RootNode,
+        progressManager: StepProgress,
+        checkCanceled: () => void
+    ): Promise<void> {
+        let foundIssues: boolean = false;
+        await this.scanDescriptorGraph(descriptorData, descriptorNode, descriptorGraph, progressManager, checkCanceled)
+            .then(descriptorWithIssues => {
+                if (descriptorWithIssues) {
+                    // Add to data and update view
+                    workspaceData.descriptorsIssuesData.push(descriptorData);
+                    root.addChildAndApply(descriptorWithIssues);
+                    foundIssues = true;
+                }
+            })
+            .catch(error => this.onFileScanError(workspaceData, root, error, descriptorData))
+            .finally(() => progressManager.onProgress());
+        if (!foundIssues) {
+            progressManager.reportProgress();
+            return;
+        }
+        await this.cveApplicableScanning(root, descriptorData, descriptorNode, progressManager.abortController)
+            .catch(err => this.onScanError(err))
+            .finally(() => progressManager.reportProgress());
+    }
+
     /**
      * Handle errors that occur during workspace scan, checks if cancele was requested.
      * @param error - the error that occur
      * @param handle - if true the error will be logged and not thrown/returned.
      * @returns -  undefined if error was handled or error otherwise
```

Comment:
```suggestion
     * Handle errors that occur during workspace scan, and check if cancellation was requested.
     * @param error - the error occurred
     * @param handle - if true the error will be logged and not thrown/returned.
     * @returns undefined if the error was handled or an error otherwise
```